wagtail/core/blocks/base.py

```diff
@@ -121,6 +127,16 @@ def set_name(self, name):
         if not self.meta.label:
             self.label = capfirst(force_str(name).replace('_', ' '))
 
+    def set_meta_options(self, options):
```

Comment:
Could this have a more specific name to make it clear it's only for the top-loevel block? eg ``set_root_block_meta_options``

Reply:
Actually I think the comment is the thing that needs updating here... initially I'd called this something like `set_options_from_field`, with the plan that the `StreamField` constructor wouldn't care about the semantics of the additional kwargs, and would just pass them here unfiltered (e.g. `blank=True` rather than `required=False`) for the block to use as it sees fit. However, that didn't work out because `StreamField` needs to pull out the StreamBlock-related kwargs before calling `super()` - so at that point it made sense for `StreamField` to translate them into StreamBlock terms (`blank` -> `required`). As a result, there's nothing field-specific here any more, so no reason why it couldn't be used for other blocks besides the top-level one.